### DISCUSSION TITLE
fzf-lua vs. telescope with telescope-fzf-native

### DISCUSSION THREAD

**Discussion by Old-Farmer** [LazyVim/LazyVim#3619] [Q&A]
Notice that fzf-lua has been introduced in extras. Is this better than telescope with telescope-fzf-native?

**Comment by dusty-phillips** [comment]
@folke I'd also like to know what the plan is here so I can match my book to it. 😅

I see that fzf-lua is a Recommended Extra, and that Telescope is installed by default *unless* you enable the fzf-lua extra. So I'm sensing that fzf-lua is preferred and hypothesize that LazyVim 13 is going to use fzf-lua by default? For my sanity as an author, I would like there to be a preferred way. I don't really have an opinion on what the preferred way is/should be, but I'd like to know what yours is so I can match it!

For the most part, I can change most of my telescope references in the book to a more generic "picker", as you've done in the code. I can think of three places I'll need to make some customizations, though:

* I suggest people install "all the recommended extras". I need to either change this to "except fzf-lua" or "especially fzf-lua" so that I can make assumptions about which picker they are using later.
* I suggest adding a keybinding to delete buffers from the Telescope buffer picker. If I instruct them to install fzf-lua, then I'll need to change this to add a keybinding to fzf-lua.
* I suggest they may want to install telescope's live-grep-args extension. I'm not sure if fzf has anything similar, so I may need to rip that section out.

The main thing I want to avoid is having conditional "if you installed fzf-lua..." throughout. I have that already with Neotree vs mini.files, and it's boring to maintain. 😆

**Reply by rubiin** [reply]
From what I read on the fzf-lua , the first difference is about performance

**Reply by rubiin** [reply]
Also if you are using fzf on terminal, its quite sensible to have it on your editor as well
More read here https://github.com/ibhagwan/fzf-lua#rationale

**Reply by AmmarCodes** [reply]
With large codebases fzf-lua is faster than Telescope with all performance tweaks applied.

I tried to force myself to use Telescope for a few days, then I switched back to fzf-lua.

**Reply by seflue** [reply]
> With large codebases fzf-lua is faster than Telescope with all performance tweaks applied.
> 
> I tried to force myself to use Telescope for a few days, then I switched back to fzf-lua.

Is the performance difference also as big when you use `telescope-fzf-native.nvim`?
